usr/src/uts/common/fs/zfs/metaslab.c

```diff
@@ -2143,6 +2473,113 @@ metaslab_distance(metaslab_t *msp, dva_t *dva)
 
 /*
  * ==========================================================================
+ * Metaslab allocation tracing facility
```

Comment:
FYI.  For zfsonlinux we conditionally removed the tracing since no imediate way to observe it.